src/inference/include/openvino/runtime/compiled_model.hpp

```diff
@@ -186,9 +186,11 @@ class OPENVINO_RUNTIME_API CompiledModel {
      * particular device.
      *
      * @param name Property key, can be found in openvino/runtime/properties.hpp.
+     * @param target_device A hardware device name, if not empty, will retrieve the target_device property via the meta
+     * plugin.
      * @return Property value.
      */
-    Any get_property(const std::string& name) const;
+    Any get_property(const std::string& name, const std::string& target_device = "") const;
```

Comment:
Sure. We actually planed to implement this feature for `set_property ` with same manner in another PR tracked by CVS-96194.

Reply:
why overload with `std::pair<std::string, ov::Property<T, mutability>>` is not enough ?

Reply:
For reusing the implementation of `get_property(const std::string& name)`, need to add a parameter to indicate from which device we want to query the property. So overloaded `get_property()` with `std::pair<std::string, ov::Property<T, mutability>>` just forward the property name and device name to original `get_porperty()` instead of reimplementing a new one.

Reply:
It would be nice to have common API for additional parameters. It can be useful for hetero case, where subgraph id can be introduced, for example, to specify concrete compiled subgraph. So instead of string parameter it would be nice to have something like this:
```
auto prop = compiled_model(ov::supported_properties, ov::device::name("CPU"));
```